lte/gateway/c/session_manager/LocalSessionManagerHandler.cpp

```diff
@@ -242,54 +242,73 @@ void LocalSessionManagerHandlerImpl::CreateSession(
-  enforcer_->get_event_base().runInEventBaseThread(
-      [this, context, response_callback, request_cpy]() {
-        SessionConfig cfg(request_cpy);
-        const std::string& imsi = cfg.get_imsi();
-        const auto& session_id  = id_gen_.gen_session_id(imsi);
-        log_create_session(cfg);
-        if (pipelined_state_ != READY) {
-          MLOG(MINFO) << "Rejecting LocalCreateSessionRequest for " << imsi
-                      << " apn=" << cfg.common_context.apn()
-                      << " since PipelineD is still setting up.";
-          send_local_create_session_response(
-              Status(grpc::UNAVAILABLE, "PipelineD is not ready"), session_id,
-              response_callback);
-          return;
-        }
-        if (!session_store_.is_ready()) {
-          MLOG(MINFO) << "Rejecting LocalCreateSessionRequest for " << imsi
-                      << " apn=" << cfg.common_context.apn()
-                      << " since SessionStore (Redis) is unavailable.";
-          send_local_create_session_response(
-              Status(grpc::UNAVAILABLE, "Storage backend is not available"),
-              session_id, response_callback);
-          return;
-        }
+  enforcer_->get_event_base().runInEventBaseThread([this, context,
+                                                    response_callback,
+                                                    request_cpy]() {
+    SessionConfig cfg(request_cpy);
+    const std::string& imsi = cfg.get_imsi();
+    const auto& session_id  = id_gen_.gen_session_id(imsi);
+    log_create_session(cfg);
+    if (pipelined_state_ != READY) {
+      MLOG(MINFO) << "Rejecting LocalCreateSessionRequest for " << imsi
+                  << " apn=" << cfg.common_context.apn()
+                  << " since PipelineD is still setting up.";
+      send_local_create_session_response(
+          Status(grpc::UNAVAILABLE, "PipelineD is not ready"), session_id,
+          response_callback);
+      return;
+    }
+    if (!session_store_.is_ready()) {
+      MLOG(MINFO) << "Rejecting LocalCreateSessionRequest for " << imsi
+                  << " apn=" << cfg.common_context.apn()
+                  << " since SessionStore (Redis) is unavailable.";
+      send_local_create_session_response(
+          Status(grpc::UNAVAILABLE, "Storage backend is not available"),
+          session_id, response_callback);
+      return;
+    }
+    const CommonSessionContext common = cfg.common_context;
+    if (common.rat_type() != TGPP_WLAN && common.rat_type() != TGPP_LTE) {
+      // We don't support outside of WLAN / LTE
+      std::ostringstream failure_stream;
+      failure_stream << "Received an invalid RAT type " << common.rat_type();
+      std::string failure_msg = failure_stream.str();
+      MLOG(MERROR) << failure_msg;
+      events_reporter_->session_create_failure(cfg, failure_msg);
+      send_local_create_session_response(
+          Status(grpc::FAILED_PRECONDITION, "Invalid RAT type"), session_id,
```

Comment:
does it make any sense to have session_id here if we don't even support this?

Reply:
ya i think the sessionid should be ignored, will change to ""